pkg/test/client.go

```diff
@@ -129,6 +132,14 @@ func Update(ctx context.Context, cl *FakeClient, obj client.Object, opts ...clie
 	currentMap["status"] = nil
 	currentMap["kind"] = nil
 	currentMap["apiVersion"] = nil
+	if currentMap["spec"] == nil {
+		currentMap["spec"] = map[string]interface{}{}
+	}
+	for key, value := range currentMap {
+		if _, exist := updatingMap[key]; !exist && value == nil {
+			updatingMap[key] = nil
```

Comment:
why not doing `delete(updateMap, key)`?

Reply:
I would need to delete it from the `currentMap` (and not `updateMap` as you suggested) because that is the one that contains the nil value. However, since I would need to check the content of the `updatingMap` anyway, then there wouldn't be any room for making the code simpler.

Reply:
ok